### PR TITLE
Improve withState in the ProfileContainer

### PR DESCRIPTION
**Pull request by tgmarinho**
Verify if my code is ok. I did attempt change to withState of recompose, for architecure more interesting with Smart -> Dumb componenets with ProfileContainer (Smart) and Profile(Dumb).

There a problem in the Alert, maybe the package future-ui-alert is not updated in npm with redux. 

Take a look! Thanks.

### CODE REVIEW

File: examples/todo/imports/ui/profile/ProfileContainer.js [hunk 1]
```diff
@@ -19,11 +21,58 @@ export const editProfileMutation = graphql(
...
 export const ProfileContainer = compose(
   editProfileMutation,
   loggedUser,
   withRouter,
-  withApollo
+  withApollo,
+  // withAlert,
+  withState('editProfileState', 'setEditProfileState', props =>
+    intialState = {
+      _id: props.loggedUser._id,
```
Comment (filipenevola) [inline examples/todo/imports/ui/profile/ProfileContainer.js]: why not only ({ ....  })

you are not declaring intialState (typo anyway, should be initialState) and not using it, just implicit returning it

File: examples/todo/imports/ui/profile/ProfileContainer.js [hunk 2]
```diff
@@ -19,11 +21,58 @@ export const editProfileMutation = graphql(
   }
 );
 
+
 const loggedUser = getLoggedUserContext();
 
+handleSubmitProfile = (editProfileState) => !!editProfileState.firstName.trim()
```
Comment (filipenevola) [inline examples/todo/imports/ui/profile/ProfileContainer.js]: is it a validation method, right? the name is wrong then.

File: examples/todo/imports/ui/profile/ProfileContainer.js [hunk 3]
```diff
@@ -19,11 +21,58 @@ export const editProfileMutation = graphql(
...
+            // TODO: I don't know how can I show the alert for user
+            console.log('Profile updated');
+            history.push('/profile');
+          })
+          .catch(error => {
+            // eslint-disable-next-line no-console
+            console.log(error);
+          });
+      } else {
+         // TODO: I don't know how can I show the alert for user
```
Comment (filipenevola) [inline examples/todo/imports/ui/profile/ProfileContainer.js]: what is the problem with alert?

File: examples/todo/imports/ui/profile/ProfileContainer.js (hunk 1 above)
Comment (tgmarinho) [reply examples/todo/imports/ui/profile/ProfileContainer.js]: Yes, I changed for it, you can see at: https://github.com/CodeFTW/future-web/commit/bc1b027604ef8d8870f7b64a267a671561bb64cd
Thanks

File: examples/todo/imports/ui/profile/ProfileContainer.js (hunk 2 above)
Comment (tgmarinho) [reply examples/todo/imports/ui/profile/ProfileContainer.js]: I think so.

File: examples/todo/imports/ui/profile/ProfileContainer.js (hunk 3 above)
Comment (tgmarinho) [reply examples/todo/imports/ui/profile/ProfileContainer.js]: old comment, I solved it with a update the package of ui-alert in 1.0.3.

**Comment by tgmarinho** [comment]
Thank you for reviewing the code, sorry but I lost the notification of yours reviews, so, I did it only now.

**Pull request merged**